CMakeLists.txt

```diff
@@ -1,187 +1,311 @@
-#------------------------------------------------------------------------------#
-# © 2021-2023. Triad National Security, LLC. All rights reserved.  This
-# program was produced under U.S. Government contract 89233218CNA000001
-# for Los Alamos National Laboratory (LANL), which is operated by Triad
-# National Security, LLC for the U.S.  Department of Energy/National
-# Nuclear Security Administration. All rights in the program are
-# reserved by Triad National Security, LLC, and the U.S. Department of
-# Energy/National Nuclear Security Administration. The Government is
-# granted for itself and others acting on its behalf a nonexclusive,
-# paid-up, irrevocable worldwide license in this material to reproduce,
-# prepare derivative works, distribute copies to the public, perform
+# ------------------------------------------------------------------------------#
+# © 2021-2023. Triad National Security, LLC. All rights reserved.  This program
+# was produced under U.S. Government contract 89233218CNA000001 for Los Alamos
+# National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S.  Department of Energy/National Nuclear Security Administration.
+# All rights in the program are reserved by Triad National Security, LLC, and
+# the U.S. Department of Energy/National Nuclear Security Administration. The
+# Government is granted for itself and others acting on its behalf a
+# nonexclusive, paid-up, irrevocable worldwide license in this material to
+# reproduce, prepare derivative works, distribute copies to the public, perform
 # publicly and display publicly, and to permit others to do so.
-#------------------------------------------------------------------------------#
-
-cmake_minimum_required(VERSION 3.14)
+# ------------------------------------------------------------------------------#
+
+# We want to stay near to latest, as more compilers and third-party libraries
+# are supported. However, uptake of new CMake releases can be slow. 20230403:
+# version 3.19 is available on all testing platforms
+cmake_minimum_required(VERSION 3.19)
+
+# Disable "in-source" builds
+if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
+  message(
+    FATAL_ERROR
+      "ERROR: Will not configure \"in-source\" builds. Create a seperate directory for building"
+  )
+endif()
```

Comment:
This is a good addition. It prevents contaminating the source tree with a bunch of files.